hangups/http_utils.py

```diff
@@ -46,6 +46,9 @@ def fetch(method, url, params=None, headers=None, cookies=None, data=None,
         else:
             error_msg = None
             break
+        finally:
+            if 'res' in locals():
+                res.close()
```

Comment:
> I can refactor the requests to use a single ClientSession. Should I create a new PR and keep this one for closing or releasing the responses?

Sure. We should probably tackle `ClientSession` separately since it might lead to other issues. :smile: